### DISCUSSION TITLE
Grid/Poly masking in MTD

### DISCUSSION THREAD

**Discussion by hertneky** [dtcenter/METplus#1522] [Configuration] [answered]
I noticed, there is no option to do grid/poly masking within the MTD tool. Is this for a reason or just not implemented for this tool yet?

I am running MTD-feature-relative for WPC work over the HRRR domain and would like to mask out snowband objects in the northwest, with a focus on running feature-relative for only the objects in the northeast. The reason for wanting to mask out the western objects is because systematic biases could be different for snowband events in different regions.

**Comment by TaraJensen** [comment, marked as answer]
Tracy,

Great question.  

While I agree, we should have consistent functionality across tools and am perplexed why we don't, I do think there's a work around while we standardize functionality. 

We just released Multivariate MODE (MvMODE) in MET v10.1.0 as part of METplus v4.1.0.  I think it would be possible to use the precip and snow fields in MvMODE to define the snowbands for use in both MODE and MTD.  Right now, the output of MvMODE is a "super-object" that is written to netCDF to be passed back into MODE.  It would be good to try passing the super-object into MTD as well.   MET documentation has been updated to show how to configure MvMODE and there is a sample MET config file in the latest release bundle.

Let us know how this goes.

**Reply by hertneky** [reply]
@TaraJensen Thanks for the response. That would be great to have included in the MTD dictionary, eventually.

With respect to MvMODE, I started playing with the default config in MET (I didn't see a METplus use-case example yet). Two issues:
1. Regridding to the forecast field, but desiring different methods for each of the fields (e.g. budget for APCP and nearest for CSNOW). Okay, through a use-case, it will be easy enough to do the regridding using regrid-data-plane 1st, then running MvMODE. So, not an issue, but just wanted to be sure this way made the most sense.
2. It seems for multivariate mode, only 1 observation file can be passed? **Edit**: I realized though trial and error that the fcst/obs_file_list for mvmode is in fact a list of files pertaining to **each field** (and **not** a list for each forecast lead, to run multiple leads).

**Reply by hertneky** [reply]
As an update on the process of using MvMODE to create snowband super-objects, I have one concern. It doesn't make sense to use precip flag (a categorical field) in MODE; doesn't make sense to apply convolution to such a field. So using precip flag to create the data mask, running single MODE (where the poly mask is applied), then MTD might make more sense? Just as a work-around until MTD has polymask capabilities.

**Reply by JohnHalleyGotway** [reply]
@TaraJensen and @hertneky I'm on discussion support today. I went hunting around to see if either of you wrote up a MET GitHub issue to add support for applying grid/poly masks to MTD, but couldn't find one.

Is this an issue that needs to be added?

**Comment by hertneky** [comment]
@JohnHalleyGotway I did not write up an issue. In my opinion, it would be helpful to have this capability in MTD. My current workaround is to just run genVxMask to do this masking. So my process list for my specific use-case leading up to running MTD looks like:

PROCESS_LIST = GenVxMask(fcst), GenVxMask(obs), GenVxMask(fcst_east), GenVxMask(obs_east), MTD

Where the 1st set does data masking on categorical snow, and the 2nd set does the east conus masking. It would be cleaner to have the options MTD_MASK_GRID/POLY, which would remove the need to run GenVxMask for a named poly or grid.

I can write up an issue for this.

**Reply by hertneky** [reply]
I wrote up an issue, this discussion can be locked (#1575)
